### PR TITLE
remove redundant functions to replace for the general has_event function

### PR DESCRIPTION
**Pull request by cjoach**
from @bfredl's idea. It removes redundant functions for a more general one and modify every appropriate calls.

I'm not sure of my docstring so if anyone have a better idea...

### CODE REVIEW

File: src/nvim/fileio.c [hunk 1]
```diff
@@ -6266,50 +6266,12 @@ int trigger_cursorhold(void)
...
-{
-  return first_autopat[(int)EVENT_INSERTCHARPRE] != NULL;
-}
-
-/// @returns true when there is an CmdUndefined autocommand defined.
-int has_cmdundefined(void)
+/// Check if "event" autocommand is defined.
+///
+/// @param event the autocommand to check
+bool has_event(int event)
```
Comment (ZyX-I) [inline src/nvim/fileio.c]: This is still FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT.

**Comment by ZyX-I** [comment]
Except for missing attributes LGTM.

**Comment by cacplate** [comment]
Can I mark it as ready and rebase? Do we need another LGTM?

**Comment by ZyX-I** [comment]
@cacplate If you rebase then it is better to squash and fix commit message in process:

    fileio: Replace some event checking functions with one has_event

**Comment by ZyX-I** [comment]
You can mark it as RDY when you feel it is ready.

File: src/nvim/edit.c [hunk 2]
```diff
@@ -1352,24 +1352,21 @@ ins_redraw (
   if (char_avail())
     return;
 
-  /* Trigger CursorMoved if the cursor moved.  Not when the popup menu is
-   * visible, the command might delete it. */
-  if (ready && (
-        has_cursormovedI()
-        ||
```
Comment (bfredl) [inline src/nvim/edit.c]: getting rid of `#ifdef` traces like this is nice :+1:

File: src/nvim/edit.c [hunk 3]
```diff
@@ -1352,24 +1352,21 @@ ins_redraw (
...
+    // Need to update the screen first, to make sure syntax
+    // highlighting is correct after making a change (e.g., inserting
+    // a "(".  The autocommand may also require a redraw, so it's done
+    // again below, unfortunately.
+    if (syntax_present(curwin) && must_redraw) {
       update_screen(0);
-    if (has_cursormovedI())
-      apply_autocmds(EVENT_CURSORMOVEDI, NULL, NULL, FALSE, curbuf);
+    }
+    if (has_event(EVENT_CURSORMOVEDI)) {
```
Comment (bfredl) [inline src/nvim/edit.c]: Let's leave it like this for now, but I wonder what (imagined?) performance problems `apply_autocmds` have that made this extra check necessary.

**Comment by bfredl** [comment]
LGTM :+1:

File: src/nvim/fileio.c [hunk 4]
```diff
@@ -6266,50 +6266,12 @@ int trigger_cursorhold(void)
...
-int has_insertcharpre(void)
-{
-  return first_autopat[(int)EVENT_INSERTCHARPRE] != NULL;
-}
-
-/// @returns true when there is an CmdUndefined autocommand defined.
-int has_cmdundefined(void)
+/// Check if "event" autocommand is defined.
+///
+/// @param event the autocommand to check
```
Comment (oni-link) [inline src/nvim/fileio.c]: '/// @returns ...' is missing.

Comment (cacplate) [inline src/nvim/fileio.c]: I thought it would be redundant: `Check if...` vs `return true if...`. Do you have a better wording in mind?

Comment (bfredl) [inline src/nvim/fileio.c]: maybe just say "return true if" on the first line and skip `@returns`

**Comment by bfredl** [comment]
looks good, please rebase

**Pull request merged**